text/0227-separate-semantic-conventions.md

```diff
@@ -0,0 +1,155 @@
+  - `resource/` - Contents of `{specification}/resource/semantic_conventions`
+  - `trace/` - Contents of `{specification}/trace/semantic_conventions`
+  - `metrics/` - Contents of `{specification}/metrics/semantic_conventions`
+  - `logs/`- Contents of `{specification}/logs/semantic_conventions`
+  - `schemas/` - A new location for [Telemetry Schemas](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/schemas/README.md)
+    to live. This directory will be hosted at
+    `https://opentelemetry.io/semantic_conventions/schemas/`
+
+Existing semantic conventions in the specification would be marked as
+deprecated, with documentation denoting the move.
+
+Additionally, if the semantic conventions eventually move to domain-specific
+director structure (e.g. `docs/{domain}/README.md`, with trace, metrics, events
+in the same file), then this can be refactored in the new repository, preserving
+git history.
+
+There will also be the following exceptions in the specification:
+
+- Semantic convetions used to implement API/SDK details will be fully specified
+  and will not be allowed to change in the Semantic Convention directory.
+  - Error/Exception handling will remain in the specification.
+  - SDK configuration interaction w/ semantic convention will remain in the
+    specification. Specifically `service.name`.
+- The specification may elevate some semantic conventions as necessary for
+  compatibility requirements, e.g. `service.instance.id` and
+  [Prometheus Compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md).
+
+These exceptions exist because:
+
+- Stable portions of the specification already rely on these conventions
+- These conventions are required to implement an SDK today.
+
+As such, the Specification should define the absolute minimum of reserved or
+required attribute names and their interaction with the SDK.
+
+## Internal details
+
+The following process would be used to ensure semantic conventions are
+seamlessly moved to their new location.
+
+- A new repository `opentelemetry/semantic_conventions` will be constructed with
+  the proposed format and necessary `Makefile` / tooling.
+- A moratorium will be placed on Semantic Convention PRs to the specififcation
+  repository. (Caveat that PRs related to this proposal would be allowed).
+- All semantic conventions in the Specification will be copied to the new
+  repository.
+- Semantic conventions in the Specification will be marked as deprecated with
+  links to the new location.
+- The Specification will be updated to require "special" conventions, like
+  `service.name` and configuration interaction.
+- Instrumentation authors will update their `SchemaURL` to
+   `https://opentelemetry.io/semantic_conventions/schemas/{semconv_version}`
+   from
+   `https://opentelemetry.io/schemas/{spec_version}`
+
```

Comment:
Currently missing: Which semantic conventions version will we start with in the new repo?
`v0.0.1`, `v1.0.0`, `v1.21.0` (the spec is currently at `v1.20.0`), ...?

Reply:
Good callout, I can add this. I was planning to drop back to a `v0.0.1` or `v0.1` while we ramp-up to HTTP stability.

Reply:
Alright if we go with this approach, we would need to use a new schema URL format anyway (https://github.com/open-telemetry/oteps/pull/227#discussion_r1143577641) or we'd run into conflicts at some point.

Reply:
Exactly.  By using a different URL we also split into a new "schema family" so compliance with this new repo is not the same as compliance with the current spec "schema family".   This gives us a chance to migrate and understand the progress of that migration.

Reply:
>Good callout, I can add this. I was planning to drop back to a v0.0.1 or v0.1 while we ramp-up to HTTP stability.

I don't understand why this decision. What is wrong with continuing the current version numbers? The Schema version numbers starting with 1.x don't imply any special stability that forces you to use 0.x to denote instability.

Reply:
I think there's a few "oddities" in the existing ecosystem we'll have to tease out and this migration is worth the effort.

E.g. the `enum` [discussion](https://github.com/open-telemetry/opentelemetry-specification/pull/3225/files#r1150119273) and its implications on the stability of existing conventions and changes required.

That said, if we just want to migrate things over with the existing version and evolve in place, I think this may wind up with *harder* to deal with breakages, but I can carve out a path for this.